### ISSUE TITLE
Possible bug in FocalLoss

### ISSUE THREAD

**Issue by dkoguciuk** [zylo117/Yet-Another-EfficientDet-Pytorch#117]
Hi @zylo117 ,

Here's what I've done:
1) I've cloned your repo - my current commit is `b62473b`
2) I've set up the dataset properly under `datasets/coco` - here you have to believe me ;) 
3) I've downloaded `efficientdet-d0.pth` and put it in `weights` folder
4) I've run evaluation successfully and got high mAP with:
`python3 coco_eval.py -p coco -c 0 -w weights/efficientdet-d0.pth`
5) I've started training on coco:
python train.py -c 0 --batch_size 8 --lr 1e-5 --num_epochs 10 --load_weights weights/efficientdet-d0.pth 

And I've got:
```
index created!
[Info] loaded weights: efficientdet-d0.pth, resuming checkpoint from step: 0
Step: 0. Epoch: 0/10. Iteration: 1/10347. Cls loss: 1.41593. Reg loss: 0.03253. Total loss: 1.44846:
```

I believe `reg_loss=0.03253` is fine, but `cls_loss=1.41593` looks suspicious. Or not?

**Comment by zylo117** [comment]
yes, suspicious.

did you run tutorial and get a decent result?

**Comment by zylo117** [comment]
it's also possible that you are suffering overfitting. try lower lr and train with `--head_only True`

**Comment by dkoguciuk** [comment]
But as you can see it's first learning iteration, so the loss is printed before any optimization step.

And yes, as stated in the 4th point, I've got the same eval results using your `efficientdet-d0.pth`. And again, it's coco (no transfer learning to another dataset). Just clean git clone of your repo.

**Comment by zylo117** [comment]
I see, I tried to replicate the issue by running with your command lines. And yes, it's almost the same as you described. 

It's not a bug, considering the community has already trained custom dataset well and so did the tutorial.

Is it too late to take back what I said? no, it's not suspicious at all. 

I ran with `--debug True` and didn't find many wrong boxes in `test/` folder, everything seems normal.

**Comment by dkoguciuk** [comment]
Yes indeed, the outputs seem normal and I've analyzed the loss function line by line and I haven't found anything wrong. I am not saying there is a bug - it's just strange for me that regression loss is 100x smaller than classification loss. 

I'll close the issue for now and will let you know if my doubts come back.
